ride_share_noor.rb

```diff
@@ -0,0 +1,192 @@
+# Copy your list from above, and in this section
+# determine what data structure each layer should have
+# trips_data (hashes)
+# rides (arrays of hashes)
+# each single ride has its own hash
+########################################################
+# Step 3: Make the data structure!
+# # Setup the entire data structure:
+# # based off of the notes you have above, create the
+# # and manually write in data presented in rides.csv
+# # You should be copying and pasting the literal data
+# # into this data structure, such as "DR0004"
+# # and "3rd Feb 2016" and "RD0022"
+
+trips_data = [{
+    driver_id: "DR0001",
+    rides: [
+        {
+            date: "3rd Feb 2016",
+            cost: 10,
+            rider_id: "RD0003",
+            rating: 3
+        },{
+            date: "3rd Feb 2016",
+            cost: 30,
+            rider_id: "RD0015",
+            rating: 4
+        },{
+            date: "5th Feb 2016",
+            cost: 45,
+            rider_id: "RD0003",
+            rating: 2
+        }]},{
+    driver_id: "DR0002",
+    rides: [
+        {
+            date: "3rd Feb 2016",
+            cost: 25,
+            rider_id: "RD0073",
+            rating: 5
+        },{
+            date: "4th Feb 2016",
+            cost: 15,
+            rider_id: "RD0013",
+            rating: 1
+        },{
+            date: "5th Feb 2016",
+            cost: 35,
+            rider_id: "RD0066",
+            rating: 3
+        }]},{
+    driver_id: "DR0003",
+    rides: [
+        {
+            date: "4th Feb 2016",
+            cost: 5,
+            rider_id: "RD0066",
+            rating: 5
+        },{
+            date: "5th Feb 2016",
+            cost: 50,
+            rider_id: "RD0003",
+            rating: 2
+        }]},{
+    driver_id: "DR0004",
+    rides: [
+        {
+            date: "4th Feb 2016",
+            cost: 10,
+            rider_id: "RD0022",
+            rating: 4
+        },{
+            date: "5th Feb 2016",
+            cost: 20,
+            rider_id: "RD0073",
+            rating: 5
+        },{
+            date: "3rd Feb 2016",
+            cost: 5,
+            rider_id: "RD0022",
+            rating: 5
+        }]}]
+
+def line
+        puts "-" * 55
+end
+
+def line2
+        puts "-" * 104
+end
+########################################################
+# Step 4: Total Driver's Earnings and Number of Rides
+
+# Use an iteration blocks to print the following answers:
+# - the number of rides each driver has given
+
+line
+puts "Number of rides each driver has given"
+line
+trips_data.map { |driver_trips|
+  puts "Driver with ID of #{driver_trips[:driver_id]} made #{driver_trips[:rides].length} rides"}
+line
+
+# - the total amount of money each driver has made
+puts "Total amount of money each driver has made"
+line
+
+trips_data.map do |driver_trips|
+  sum = driver_trips[:rides].sum do |ride|
+    ride[:cost]
+  end
+  puts "Driver with ID of #{driver_trips[:driver_id]} earned $#{sum} "
+  end
```

Comment:
Check your indentation here, and in general we use `map`, when we want to return a new array which is the old one with every element modified in some way.

You are using this like a `.each` loop.

I suggest instead making a method.

```suggestion
def earnings
  earnings_list = trips_data.map do |driver_trips|
    sum = driver_trips[:rides].sum do |ride|
      ride[:cost]
    end
    
    { driver: driver_trips[: driver_id], earnings: sum }
  end
  
  return earnings_list
end

driver_earnings  = earnings(trips_data)
driver_earnings.each do |driver|
  puts "Driver with ID of #{driver[:driver_id]} earned $#{driver[:earnings]} "
end
```

Reply:
Oh I did used the .map alot like .each! Thank you Chris! the method you wrote gives me lots of ideas of how to improve my coding!